src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,574 @@
+        if (isset($yamlDatas['parameters'])) {
+            $this->addNodeParameters($yamlDatas['parameters']);
+        }
+
+        if (isset($yamlDatas['imports'])) {
+            $this->addNodeImports($yamlDatas['imports']);
+        }
+
+        if (isset($yamlDatas['services'])) {
+            $this->addNodeServices($yamlDatas['services']);
+        }
+
+        sort($this->useStatements);
+        foreach ($this->useStatements as $className) {
+            $this->configurationFactory->addStmt(
+                $this->factory->use($className)
+            );
+        }
+
+        // remove the last carriage return "\n" if exists
+        $this->stmts[array_key_last($this->stmts)]->parts[0] = rtrim($this->stmts[array_key_last($this->stmts)]->parts[0], "\n");
+
+        return $this->stmts;
+    }
+
+    private function addNodeParameters($datasParameters)
+    {
+        $this->addLineStmt('// Put parameters here that don\'t need to change on each machine where the app is deployed.');
+        $this->addLineStmt('// https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration.');
+        $this->addLineStmt('$parameters = $configurator->parameters();', true);
+
+        foreach ($datasParameters as $keyParameter => $values) {
+            $this->addLineStmt(
+                $this->createParameterCallMethod($keyParameter, $values, $isParameter = true), true
+            );
+        }
+    }
+
+    private function addNodeImports($datasImports)
+    {
+        $import = '$configurator->import('.$this->createListFromArray($datasImports[0]).');';
+
+        $this->addLineStmt($import, true);
+    }
+
+    private function addNodeServices($servicesDatas)
+    {
+        $this->addLineStmt('$services = $configurator->services();', true);
+
+        foreach ($servicesDatas as $serviceKey => $serviceValues) {
+            if ('_' === $serviceKey[0]) {
+                $this->add_ConfigurationNodes(ltrim($serviceKey, '_'), $serviceValues);
+            } elseif (class_exists($serviceKey)) {
+                $className = $this->addUseStatementIfNecessary($serviceKey);
+
+                if (null === $serviceValues) {
+                    $this->addLineStmt('$services->set('.$className.');', true);
+                }
+
+                if (\is_string($serviceValues) && '@' === $serviceValues[0]) {
+                    $this->addLineStmt(
+                        '$services->alias('.$className.', \''.substr($serviceValues, 1).'\');', true
+                    );
+                }
+
+                if (\is_array($serviceValues)) {
+                    $this->addLineStmt('$services->set('.$className.')');
+
+                    foreach ($serviceValues as $subKey => $subValue) {
+                        $this->addFluentCallMethodNodes($subKey, $serviceValues, $subKey);
+                    }
+                    $this->addLineStmt(';', true);
+                }
+            } elseif (strstr($serviceKey, '.')) {
```

Comment:
We can use a strategy here to avoid `else`... `else` generally makes code less readable. Basically, do this:

```php
if ('_' === $serviceKey[0]) {
    $this->add_ConfigurationNodes(ltrim($serviceKey, '_'), $serviceValues);

    continue;
}

if (class_exists($serviceKey)) {
    $className = $this->addUseStatementIfNecessary($serviceKey);

    // rest of logic

    continue;
}

if (strstr($serviceKey, '.')) {
    // ...
}

// ...
```

Reply:
Yeah, I was gonna ask you, thanks!